pkg/discovery/discovery.go

```diff
@@ -114,3 +120,59 @@ func (td *tidbDiscovery) Discover(advertisePeerUrl string) (string, error) {
 	delete(currentCluster.peers, podName)
 	return fmt.Sprintf("--join=%s", strings.Join(membersArr, ",")), nil
 }
+
+func (td *tidbDiscovery) DiscoverDM(advertisePeerUrl string) (string, error) {
+	td.lock.Lock()
+	defer td.lock.Unlock()
+
+	if advertisePeerUrl == "" {
+		return "", fmt.Errorf("dm advertisePeerUrl is empty")
+	}
+	klog.Infof("dm advertisePeerUrl is: %s", advertisePeerUrl)
+	strArr := strings.Split(advertisePeerUrl, ".")
+	if len(strArr) != 4 {
+		return "", fmt.Errorf("dm advertisePeerUrl format is wrong: %s", advertisePeerUrl)
+	}
+
+	podName, peerServiceName, ns := strArr[0], strArr[1], strArr[2]
+	dcName := strings.TrimSuffix(peerServiceName, "-dm-master-peer")
+	podNamespace := os.Getenv("MY_POD_NAMESPACE")
+	if ns != podNamespace {
+		return "", fmt.Errorf("dm the peer's namespace: %s is not equal to discovery namespace: %s", ns, podNamespace)
+	}
+	dc, err := td.cli.PingcapV1alpha1().DMClusters(ns).Get(dcName, metav1.GetOptions{})
```

Comment:
The role for discovery in the `tidb_discovery_manager.go` needs to update.

Reply:
The role added in the above file means that the discovery can only list the DMClusters in the same ns, so the ns check in the previous commit should be OK.